lib/puppet/util/log/destinations.rb

```diff
@@ -47,7 +47,7 @@ def handle(msg)
   require 'fileutils'
 
   def self.match?(obj)
-    Puppet::Util.absolute_path?(obj)
+    obj.is_a?(String) && Puppet::Util.absolute_path?(obj)
```

Comment:
Can `obj` be a `Pathname`?

Reply:
`Pathname` doesn't respond to `=~` so I would assume that couldn't have worked.